### ISSUE TITLE
BundleDict assignment doesn't work? 

### ISSUE THREAD

**Issue by arokem** [yeatmanlab/pyAFQ#956]
From [our pARC/VOF example](https://github.com/yeatmanlab/pyAFQ/blob/master/examples/vof_example.py): 

```
import AFQ.api.bundle_dict as abd
import AFQ.data.fetch as afd
from AFQ.api.group import GroupAFQ
from AFQ.definitions.image import RoiImage
import AFQ.utils.streamlines as aus

import os.path as op

afd.organize_stanford_data(clear_previous_afq=True)

bundle_names = ["ARC_L", "ARC_R", "pARC_L", "pARC_R", "VOF_L", "VOF_R"]
bundle_dict = abd.BundleDict(bundle_names)

# you can change these if you want
bundle_dict["VOF_R"]["primary_axis_percentage"] = 50
bundle_dict["VOF_L"]["primary_axis_percentage"] = 50
```

But then: 

```
print(bundle_dict["VOF_R"]["primary_axis_percentage"])
```

```
40
```

Am I missing something?

**Comment by 36000** [comment]
It's being caused by this line here:
https://github.com/yeatmanlab/pyAFQ/blob/7bf8b2527441262aee4b26a7eeac6798a021c9b2/AFQ/api/bundle_dict.py#L338

In other words, bundle_dict["VOF_R"] returns a copy. This is because, by default, the bundle dict stores the path to ROIs and  then when you access a particular entry, it loads and resamples the ROI before returning. The bundle dict still only stores the path after this. If it were important to be able to change bundle definitions after instantiation, we could change this and have a separate get function for loading/resample. But I like bundle dictionary entries being immutable. My vote is to remove these lines from the example (because they aren't true) and then clarify in the documentation that you cannot change bundle dictionary entries after instantiation.

**Comment by arokem** [comment]
It makes sense for it to be immutable but maybe we should raise
when `__setitem__` is called after initialization? We can maybe do this by
replacing `__setitem__` with a custom version at the end of the `__init__`
function.

**Comment by 36000** [comment]
For example, if I add this to the pyAFQ code:
```
from types import MappingProxyType as ReadOnlyDictProxy

def __getitem__():
    ...
    _item = ReadOnlyDictProxy(self._dict[key].copy())
    ...
    return _item
```
Then the example throws this error:
```
TypeError: 'mappingproxy' object does not support item assignment
```

**Comment by 36000** [comment]
Resolved in #953
